currency/rate_fetcher.go

```diff
@@ -0,0 +1,61 @@
+package currency
+
+import (
+	"context"
+	"errors"
+)
+
+type Currency string
+
+var ErrCurrencyNotHandled = errors.New("currency: Currency not handled")
+
+type RateFetcher interface {
+	BaseCurrency() Currency
+	Rate(context.Context, Currency) (float64, error)
+}
+
+type Money struct {
+	Cents    int
```

Comment:
No sure it is good to think in cents since a few currency don't have this notion. imo you  would be better off with a float64 and play with the precision for the display.

Reply:
As far as know, there is no monetary values that needs more than a precision a two. It let's all the other bits to be used to encode large values.

Furthermore it is the way we encode it already in our system (money, currency) hence it will be easier to use